File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res;
        int i=0, j = numbers.size() - 1,sum = 0;
        while(i<=j){
            sum = numbers[i] + numbers[j];
            if(sum == target){
                res.push_back(++i);
                res.push_back(++j);
                return res;
            } else if(sum < target){
                i += 1;
            } else {
                j -= 1;
            }
        }
        return res;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = len(numbers) - 1
        while(i<=j):
            sum = numbers[i] + numbers[j]
            if(sum == target):
                return [i+1,j+1]
            elif(sum < target):
                # here we should increase the sum value to reach target so we increase the slow pointer
                i += 1
            else:
                j -= 1
            